### ISSUE TITLE
Default policy server is missing KUBEWARDEN_ENABLE_METRICS env

### ISSUE THREAD

**Issue by kravciak** [kubewarden/ui#287]
### Current Behavior

Kubewarden defaults chart is missing `KUBEWARDEN_ENABLE_METRICS` env variable when telemetry is enabled on controller.
This variable is set when I create custom policy server.

There are 2 ways how to create policy server.

1. default one -> install rancher-kubewarden-defaults chart by `InstalChart button` shown in policy servers overview info box
User has to edit yaml and and manually enable telemetry to get any metrics.

```
policyServer:
  env:
    - name: KUBEWARDEN_LOG_LEVEL
      value: info
  telemetry:
    enabled: false
...
```

2. custom policy server -> go to policy server list and click `Create` button
User do not need to change anything, metrics are enabled by env variable
```
spec:
  env:
    - name: KUBEWARDEN_ENABLE_METRICS
      value: '1'
    - name: KUBEWARDEN_LOG_FMT
...
```

### Expected Behavior

When telemetry is enabled on controller then it's automatically enabled also on new policy servers (both default and custom)

[Docs](https://docs.kubewarden.io/operator-manual/ui-extension/metrics) says it's required only to enable telemetry on kubewarden-controller, it does not mention policy-servers. We should either:
- add requirement to enable telemetry on existing policy servers
- enable it automatically when user edit kubewarden-controller resource

**Comment by jvanz** [comment]
I agree that we have, at least, a docs problem. However, afaics, I believe that controller does not enable metric by default in the policy servers in purpose. Therefore, users can customize their policy servers individually.

I'm not sure if we should propagate the telemetry config of the controller in policy servers. At least, not by default. I think this can cause issues for people already using policy servers with and without telemetry enabled.

**Comment by jvanz** [comment]
tldr: I think we should update the docs to explicit said that telemetry should be enable per policy server  and show how to do it.

In other hand, if we want to make some config to facilitate users experience, we can create a flag like "follow controller telemetry" which tells the controller to add the annotations and env var variable based on its own configuration.

**Comment by viccuad** [comment]
Did a [wide search](https://github.com/search?q=org%3Akubewarden+KUBEWARDEN_ENABLE_METRICS&type=code).

The controller reconciles the policy-server service, and appends the port, if metrics is enabled for the controller.
The enablement of metrics in the PolicyServers is left per PolicyServer, via the env var.

In the case of the default PolicyServer from `kubewarden-defaults`:

https://github.com/kubewarden/helm-charts/blob/70251a826537b2644f698efaae3676b1cb5b7926/charts/kubewarden-defaults/templates/policyserver-default.yaml#L28-L31

We do enable `KUBEWARDEN_ENABLE_METRICS` if `policyServer.telemetry.enabled` in `kubewarden-defaults`. With this should be enough, and if not, it sounds like a bug. 


On the UI side, it seems that we always enable telemetry for the default PolicyServer, regardless if it is enabled in the controller, or not:

https://github.com/kubewarden/ui/blob/main/pkg/kubewarden/models/policies.kubewarden.io.policyserver.js#L14

I don't think we should be doing that, as at a minimum, the PolicyServer is going to have warnings because it can't connect to the endpoint I suppose.

From my view, we should:
- Confirm telemetry is enabled if `policyServer.telemetry.enabled` in `kubewarden-defaults`.
- Indeed document that one should set `policyServer.telemetry.enabled` in `kubewarden-defaults`.
- Default to disabled telemetry when creating PolicyServers in UI
- If telemetry option is enabled in UI, configure PolicyServer's `spec.annotations."sidecar.opentelemetry.io/inject"` (which seems to be currently missing?) and `spec.env.KUBEWARDEN_ENABLE_METRICS`.

**Comment by jvanz** [comment]
> The controller reconciles the policy-server service, and appends the port, if metrics is enabled for the controller.

Not really. The controller appends the metrics port if the metrics is enabled for the [policy server](https://github.com/kubewarden/kubewarden-controller/blob/353f17d0c3f20666a5b50b9dcb1ffcd8c6ee2827/internal/pkg/admission/policy-server-service.go#L73). Thus, different policy servers can or cannot have telemetry enabled. Which I think is a good thing

Rereading this issue and @viccuad's comment I've got confused. My first though is that the @kravciak expected that the controller will add the telemetry configuration on every new policy server created if the controller has telemetry enabled. Independently of the configuration of the policy server deployed.  Otherwise, telemetry will not work properly because it will miss the policy servers data. Which makes sense to me. In the end of the day, policy data is the most valuable data available. However,  the controller metrics should be available in the telemetry system. But I'm not sure what are  the metrics/traces we have in the controller now.

However, as I said before, this is working as "expected". If we want that the controller enable telemetry in the policy servers by default, this seems a feature request to me. Not a bug in the sense that it is not doing what it is suppose to do. If this is the case, this is a change in the controller not in the UI.

**Comment by kravciak** [comment]
Agree with Victor here.
 - we don't document that user have to enable telemetry on policy server (if we don't want to do it automatically)
- enablement of telemetry is not consistent
  - default policy server does not inherit telemetry from controller
  - secondary policy servers inherit telemetry

> From my view, we should:
> 
> * Confirm telemetry is enabled if `policyServer.telemetry.enabled` in `kubewarden-defaults`.
> * Indeed document that one should set `policyServer.telemetry.enabled` in `kubewarden-defaults`.
> * Default to disabled telemetry when creating PolicyServers in UI
> * If telemetry option is enabled in UI, configure PolicyServer's `spec.annotations."sidecar.opentelemetry.io/inject"` (which seems to be currently missing?) and `spec.env.KUBEWARDEN_ENABLE_METRICS`.

**Comment by flavio** [comment]
I have the feeling the enablement of the telemetry is scattered in 3 different places: controller, helm chart and UI. I think we should have one single point that manages that to have a consistent and predictable behavior.

What if we do something like that:

* When the controller's telemetry is enabled: telemetry is enabled for all the PolicyServer objects that are reconciled by it. If the user wants to disable the telemetry of a specific PolicyServer, we can implement this feature; for example, we could document that the user must set a certain environment variable inside of the PolicyServer definition. However I would not waste time on that right now, but only do that if there's an actual business case
* Helm chart: we keep the enable telemetry variable we have. When this is set, telemetry is enabled inside of the controller. We leave PolicyServer telemetry to be handled by the controller (see previous point)
* UI: we expose the "enable telemetry" flag of the helm chart. When creating a new PolicyServer via the UI we do not show any telemetry option at all, this is handled by the controller (see 1st point)

I hope I was able to express myself properly :sweat_smile:

**Comment by viccuad** [comment]
So then, that means,
- controller: When `--enable-metrics`, make it set the correct things on all PolicyServers (`spec.annotations."sidecar.opentelemetry.io/inject"`, `spec.env.KUBEWARDEN_ENABLE_METRIC`).
- `kubewarden-controller` helm chart: Keep the setting `Values.telemetry.enabled` that triggers the previous point.
- `kubewarden-defaults` helm chart: Remove setting  `Values.policyServer.telemetry.enabled`, not needed.
- UI: expose "enable telemetry" flag of the `kubewarden-controller` chart.
- UI: don't add env vars by hand on custom PolicyServer creation, as it is done by the controller.

**Comment by viccuad** [comment]
Started looking on implementing the suggestion in the controller. About the first point:

>   * controller: When `--enable-metrics`, make it set the correct things on all PolicyServers
> (`spec.annotations."sidecar.opentelemetry.io/inject"`, `spec.env.KUBEWARDEN_ENABLE_METRIC`).

If we follow that, then we cannot disable metrics per PolicyServer, the reconcile loop would add the env var and sidecar again.

For PolicyServers with telemetry, we need:
- Add sidecar via `spec.annotations."sidecar.opentelemetry.io/inject"` (currently set for default PolicyServer in `kubewarden-defaults`, unset for rest)
- Enable telemetry in policy-server via `spec.env.KUBEWARDEN_ENABLE_METRIC` (currently set for default PolicyServer in `kubewarden-defaults`, unset for rest, set for all in UI)
- Add port to PolicyServer service where the metrics are exported (currently added for all PolicyServers by the controller)

If we want only 1 place to enable/disable the metrics feature, in the controller, then I don't see a simple UX to be able to disable metrics per PolicyServer. Adding a filter list of PolicyServers without metrics in the controller feels like overkill.

Are we ok with a "metrics everywhere or nowhere" approach by enabling them in the controller?

**Comment by kravciak** [comment]
I would go for easier solution, just to have it consistent / documented. Both options sound fine.

**Comment by viccuad** [comment]
documenting, we decided on going with Option 1, and not allowing for now to disable telemetry per PolicyServer.

**Comment by jvanz** [comment]
Fixed and released.
